File: model/src/main/java/io/github/chriszhong/model/organization/event/SpecificationGoalEventFactory.java
package io.github.chriszhong.model.organization.event;

import io.github.runtimemodels.chazm.entity.SpecificationGoal;

/**
 * The {@linkplain SpecificationGoalEventFactory} interface defines the API for constructing {@linkplain SpecificationGoalEvent}s.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
@FunctionalInterface
public interface SpecificationGoalEventFactory {

	/**
	 * Constructs an {@linkplain SpecificationGoalEvent}.
	 *
	 * @param category
	 *            the {@linkplain EventCategory}.
	 * @param goal
	 *            the {@linkplain SpecificationGoal}.
	 *
	 * @return a {@linkplain SpecificationGoalEvent}.
	 */
	SpecificationGoalEvent build(EventCategory category, SpecificationGoal goal);

}

File: chazm-model/src/main/kotlin/runtimemodels/chazm/model/entity/SpecificationGoalEventFactory.kt
package runtimemodels.chazm.model.entity

import runtimemodels.chazm.api.entity.SpecificationGoal
import runtimemodels.chazm.model.event.EventType

/**
 * The [SpecificationGoalEventFactory] interface defines the API for constructing [SpecificationGoalEvent]s.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
@FunctionalInterface
interface SpecificationGoalEventFactory {
    /**
     * Constructs an [SpecificationGoalEvent].
     *
     * @param category the [EventType].
     * @param goal     the [SpecificationGoal].
     * @return a [SpecificationGoalEvent].
     */
    fun build(category: EventType, goal: SpecificationGoal): SpecificationGoalEvent
}
